### PR TITLE
Update kubernetes-client-api to 5.4.1

### PR DESCRIPTION
**Pull request by Vlatombe**
Downstream of https://github.com/jenkinsci/kubernetes-client-api-plugin/releases/tag/kubernetes-client-api-5.4.1

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by Vlatombe** [comment]
Not sure what has changed but I'm getting similar failures on master.

File: pom.xml [hunk 1]
```diff
@@ -144,7 +144,7 @@
       <dependency>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>jackson2-api</artifactId>
-        <version>2.10.2</version>
+        <version>2.12.1</version>
```
Comment (Vlatombe) [inline pom.xml:147]:
```suggestion
        <version>2.12.3</version>

```

File: pom.xml [hunk 2]
```diff
@@ -164,7 +164,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>kubernetes-client-api</artifactId>
-      <version>4.6.0-2</version>
+      <version>5.4.1-beta-1</version>
```
Comment (Vlatombe) [inline pom.xml:167]:
```suggestion
      <version>5.4.1</version>

```

**Comment by Vlatombe** [comment]
This pull request should be merged and released in order to achieve compatibility with [kubernetes-client-api 5.4.1](https://github.com/jenkinsci/kubernetes-client-api-plugin/releases/tag/kubernetes-client-api-5.4.1).

**Comment by topikachu** [comment]
I have an issue when start Jenkins
```
java.lang.NoSuchMethodError: io.fabric8.kubernetes.client.dsl.FilterWatchListDeletable.watch(Ljava/lang/String;Ljava/lang/Object;)Ljava/lang/Object;
	at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.KubernetesCredentialProvider.startWatchingForSecrets(KubernetesCredentialProvider.java:128)
Caused: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.init.TaskMethodFinder.invoke(TaskMethodFinder.java:104)
```

Kubernetes Client API Plugin 5.4.1
Kubernetes Credentials Provider 0.18-1
Jenkins 2.289.1

**Comment by Vlatombe** [comment]
@topikachu you should revert Kubernetes Client API Plugin until this plugin catches up.

**Comment by Vlatombe** [comment]
@jtnord fyi

**Review by jtnord — APPROVED** [review]

**Comment by jtnord** [comment]
@Vlatombe Jenkins reports test failures

**Comment by Vlatombe** [comment]
> @Vlatombe Jenkins reports test failures

https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/pull/54#issuecomment-858628876

**Comment by jtnord** [comment]
https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8202343

Oh clucking bell.....

**Comment by jtnord** [comment]
@Vlatombe  I ran the tests locally on master and they pass, 2 tests fail with this PR.  (just use a JDK older than 8u291)

```
[ERROR] Tests run: 25, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.02 s <<< FAILURE! - in com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtilsTest
[ERROR] com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtilsTest.getNonNullSecretDataWithNoData  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NullPointerException
        at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtils.getNonNullSecretData(SecretUtils.java:200)
        at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtilsTest.getNonNullSecretDataWithNoData(SecretUtilsTest.java:189)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)

[ERROR] com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtilsTest.getNonNullSecretDataWithNoMappedData  Time elapsed: 0 s  <<< ERROR!
java.lang.NullPointerException
        at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtils.getNonNullSecretData(SecretUtils.java:202)
        at com.cloudbees.jenkins.plugins.kubernetes_credentials_provider.SecretUtilsTest.getNonNullSecretDataWithNoMappedData(SecretUtilsTest.java:202)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)

```

**Review by jtnord — CHANGES_REQUESTED** [review]
2 new test failures unrelated to the TLS negotiotion/

**Comment by Vlatombe** [comment]
> 2 new test failures unrelated to the TLS negotiotion/

Fixed. Indeed, kubernetes-client made some changes in their fluent api generator that needed adjustment.

**Comment by jtnord** [comment]
thanks @Vlatombe

**Comment by jtnord** [comment]
released [kubernetes-credentials-provider-0.20](https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/releases/tag/kubernetes-credentials-provider-0.20)

**Pull request merged**